### PR TITLE
Add release notes for data prepper release v2.10

### PR DESCRIPTION
**Pull request by sb2k16**
### Description
This PR is to add release notes for Data Prepper release v2.10.
 
### Issues Resolved
Resolves #3371
 
### Check List
- [X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: release/release-notes/data-prepper.release-notes-2.10.0.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+### Features
+
+* Kafka source: support SASL/SCRAM mechanisms ([#4241](https://github.com/opensearch-project/data-prepper/issues/4241))
+* OpenSearch Bulk API Source ([#248](https://github.com/opensearch-project/data-prepper/issues/248))
+* Support AWS Kinesis Data Streams as a Source ([#1082](https://github.com/opensearch-project/data-prepper/issues/1082))
+
+### Enhancements
+
+* Support OpenTelemetry logs in S3 source ([#5028](https://github.com/opensearch-project/data-prepper/issues/5028))
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-2.10.0.md:15]: Let's move this to the features section. Supporting OTel logs in S3 opens up interesting possibilities.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: release/release-notes/data-prepper.release-notes-2.10.0.md (hunk 1 above)
Comment (sb2k16) [reply release/release-notes/data-prepper.release-notes-2.10.0.md]: Thanks @dlvenable. I have addressed this.

**Review by dlvenable — APPROVED** [review]
Thank you @sb2k16 for adding these release notes!

File: release/release-notes/data-prepper.release-notes-2.10.0.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+## 2024-10-14 Version 2.10.0
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-2.10.0.md:1]:
```suggestion
## 2024-10-15 Version 2.10.0
```

Please update the date for today's release

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: release/release-notes/data-prepper.release-notes-2.10.0.md (hunk 2 above)
Comment (sb2k16) [reply release/release-notes/data-prepper.release-notes-2.10.0.md]: Thanks @dlvenable. I have updated the date.

**Review by dlvenable — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Pull request merged**